### DISCUSSION TITLE
Developer contributor Guide

### DISCUSSION THREAD

**Discussion by appsec-airito** [projectdiscovery/nuclei#1298] [General]
Is there a link that provides - some nuclei specific guidance for coding?
I got a few issues with the code at: https://github.com/projectdiscovery/nuclei/pull/1294

I want to follow up and correct them

**Comment by appsec-airito** [comment]
Eg:
I looked into the master and contributed a code there- but was asked to merge with the dev branch. The dev branch is not in sync with master. Hence the issue with lint. If that's the case I'm facing than probably others are merging to master as well instead of going through dev.

So if there is something current that I can follow it will be helpful

**Comment by appsec-airito** [comment]
realized the reason, did changes to the dev branch and requested a PR. But would still like to go through any guidance that can help out. For instance I think some of the existing test cases are wrong- so should I write new test cases?

**Comment by ehsandeep** [comment]
@appsec-airito thanks for pointing this, soon we will add contributing.md to this project.

**Reply by MetzinAround** [reply]
https://github.com/projectdiscovery/.github/blob/6677ea6e93774acd6ea78f0bac380300cd10630e/CONTRIBUTING.md#L4
